internal/dashboard/workspace_network_handler.go

```diff
@@ -43,25 +35,26 @@ func (s *Server) PutNetworkRule(ctx context.Context, userId string, workspaceNam
 	var err error
 	ws.Spec.Network, err = wsnet.UpsertNetRule(ws.Spec.Network, netRule)
 	if err != nil {
-		res.Message = err.Error()
-		log.Error(err, res.Message, "userid", user.Name, "workspace", ws.Name, "netRuleName", networkRuleName)
-		return dashv1alpha1.Response(http.StatusBadRequest, res), nil
+		message := err.Error()
+		log.Error(err, message, "userid", userId, "workspace", ws.Name, "netRuleName", networkRuleName)
+		return ErrorResponse(http.StatusBadRequest, message)
 	}
 
 	log.DebugAll().Info("NetworkRule upserted", "ws", ws, "namespace", ws.Namespace, "netRule", netRule)
 	log.DebugAll().PrintObjectDiff(before, ws)
 
 	if equality.Semantic.DeepEqual(before, ws) {
```

Comment:
`DeepEqual(before, ws)` always returns false in production.
`TargetPortNumber` and `Host` are always different.
```
     :
                Network: []v1alpha1.NetworkRule{
                        {
                                PortName:         "nw-1",
                                PortNumber:       3434,
                                HTTPPath:         "/",
-                               TargetPortNumber: &39427,
+                               TargetPortNumber: nil,
-                               Host:             &"aa-test1-cosmo-user-xxxx.domain",
+                               Host:             nil,
                                Group:            &"aa",
                                Public:           false,
                        },
                        {PortName: "nw-3", PortNumber: 222, HTTPPath: "/", TargetPortNumber: &222, ...},
                },
```

Reply:
You are correct.